src/Http/Middleware/CheckPermissions.php

```diff
@@ -0,0 +1,51 @@
+<?php
+
+namespace Marketredesign\MrdAuth0Laravel\Http\Middleware;
+
+use App;
+use Auth0\Login\Auth0Service;
+use Closure;
+use Exception;
+use Illuminate\Http\Request;
+use Illuminate\Support\Facades\Auth;
+use Illuminate\Support\Facades\Log;
+
+class CheckPermissions
+{
+    /**
+     * Validate the user is logged in, optionally requiring some permission.
+     *
+     * @param Request $request - Illuminate HTTP Request object.
+     * @param Closure $next - Function to call when middleware is complete.
+     * @param string|null $permissionRequired - Optional, Oauth permission required
+     *
+     * @return mixed
+     * @throws Exception
+     */
+    public function handle(Request $request, Closure $next, string $permissionRequired = null)
+    {
+        // Make sure the user is logged in
+        if (!Auth::check()) {
+            return redirect()->route('login');
+        }
+
+        // Fetch the access token of the user
+        $token = Auth::user()->getAuthPassword();
```

Comment:
To me it doesn't really make sense to use the access token for this purpose. The access token should be used to access resources on behalf of the user. What we are doing here is checking the permissions of the logged-in user, which should be performed using the ID token instead, right? (following the OpenID connect protocol)